### PR TITLE
New Typography Component

### PR DESCRIPTION
**Pull request by AtelyPham**
- feat: setup new `Typography` component
- chore: add new fonts
- refactor: move custom components to tailwind component layer
- fix: wrong classNames
- chore: make use new Typography component for page title
- refactor: separate `css` files

## Summary of changes
- Add new Typography component for the Webb UI Kit
- Separate the tailwindcss config file into multiple CSS files

### Reference issue to close (if applicable)
- Closes https://github.com/webb-tools/webb-dapp/issues/505

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### Screenshot

CleanShot 2022-09-02 at 15 44 35@2x
CleanShot 2022-09-02 at 15 46 54@2x

### CODE REVIEW

File: packages/webb-ui-components/src/typograhy/Typography/Typography.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  body4: 'p' as const,
+  mono1: 'span' as const,
+  mono2: 'span' as const,
+};
+
+/**
+ * The Webb Typography component
+ *
+ * Props:
+ * - `variant`: Represent diffferent variants of the component
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/Typography/Typography.tsx:25]:
```suggestion
 * - `variant`: Represents different variants of the component
```

File: packages/webb-ui-components/src/typograhy/Typography/Typography.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * - `component`: The html tag (default: same as `variant` prop)
+ * - `fw`: Represent the **font weight** of the component (default: `left`)
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/Typography/Typography.tsx:27]:
```
* - `fw`: Represent the **font weight** of the component (default: `left`)
```
@AtelyPham I don't think the default is left here?

File: packages/webb-ui-components/src/typograhy/types.d.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+export type BodyVariant = 'body1' | 'body2' | 'body3' | 'body4';
+
+export type MonospaceVariant = 'mono1' | 'mono2';
+
+export type WebbTypographyVariant = HeadingVariant | BodyVariant | MonospaceVariant;
+
+/**
+ * Properties of typography component
+ * - `variant`: Represent diffferent variants of the component
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/types.d.ts:19]:
```suggestion
 * - `variant`: Represents different variants of the component
```

File: packages/webb-ui-components/src/typograhy/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * - `component`: The html tag (default: same as `variant` prop)
+ * - `fw`: Represent the **font weight** of the component (default: `left`)
+ * - `ta`: Text align (default: `inherit`)
+ * - `darkMode`: Control component dark mode display in `js`, leave it's empty if you want to control dark mode in `css`
+ */
+export interface WebbTypographyProps<TypoVariant = WebbTypographyVariant> extends TypographyBaseProps {
+  /**
+   * Represent diffferent variants of the component
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/types.d.ts:27]:
```suggestion
   * Represents different variants of the component
```

File: packages/webb-ui-components/src/typograhy/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  if (isMonospaceVariant(variant) && fontWeight === 'semibold') {
+    return `font-bold`;
+  } else {
+    return `font-${fontWeight}` as const;
+  }
+}
+
+/**
+ * Assert the typography variant to heading variant
+ * @param variant Represent the value to check if it is the heading varirant
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/utils/index.ts:34]:
```suggestion
 * @param variant Represents the value to check if it is the heading variant
```

File: packages/webb-ui-components/src/typograhy/utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ * @returns Whether the typography variant is heading variant
+ */
+export function isHeadingVariant(variant: WebbTypographyVariant): variant is HeadingVariant {
+  const headingKeys = ['h1', 'h2', 'h3', 'h4', 'h5'];
+  return headingKeys.indexOf(variant) !== -1;
+}
+
+/**
+ * Assert the typography variant to monospace variant
+ * @param variant Represent the value to check if it is the monospace varirant
```
Comment (dutterbutter) [inline packages/webb-ui-components/src/typograhy/utils/index.ts:44]:
```suggestion
 * @param variant Represent the value to check if it is the monospace variant
```

**Review by nepoche — APPROVED** [review]

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**